### PR TITLE
Build AOT and test targets, generate FARs when building Fuchsia

### PR DESCRIPTION
**Pull request by dnfield**
Right now my local workflow involves running 3 or 4 different scripts to achieve this - this is more analogous to what we do on other platforms, e.g. creating the flutter.jar and the test jar files when we build Android.

Since flutter_runner_tests and the relevant far are now built by default, we also don't need the shell script or the special step in CI to do it.

Once this lands, we can also simplify the LUCI recipe for Fuchsia related things that is trying to do similar work.

### CODE REVIEW

File: tools/fuchsia/fuchsia_archive.gni [hunk 1]
```diff
@@ -2,10 +2,10 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-# Creates a package dir that we will them use pm to package.
+# Creates a Fuchsia archive (.far) file using PM from the Fuchsia SDK.
 #
 # This currently ignores the CMX files and does minimal validation.
```
Comment (dnfield) [inline tools/fuchsia/fuchsia_archive.gni]: @iskakaushik does this need a TODO? Is it still even true?

**Comment by iskakaushik** [comment]
@dnfield don't we need to change the build_fuchsia python script to no longer invoke gen_package and other stuff?

File: tools/fuchsia/fuchsia_archive.gni [hunk 2]
```diff
@@ -67,12 +67,37 @@ template("package_dir") {
...
     outputs = copy_outputs
     testonly = pkg_testonly
   }
+
+  action(target_name) {
+    script = "$flutter_root/tools/fuchsia/gen_package.py"
+    deps = pkg_dir_deps + [ ":${target_name}_dir" ]
+    sources = copy_outputs
+    inputs = [
+      "$flutter_root/tools/fuchsia/gather_flutter_runner_artifacts.py",
```
Comment (iskakaushik) [inline tools/fuchsia/fuchsia_archive.gni]: shouldn't inputs be the same as inputs for `package_dir`

Comment (dnfield) [reply tools/fuchsia/fuchsia_archive.gni]: This was neeeded because the gen_package script imported that.  It's not needed anymore.

**Pull request merged**